diozero-core/src/main/java/com/diozero/devices/AHT21.java

```diff
@@ -0,0 +1,204 @@
+
+    /**
+     * Reset registry.
+     * <b>Implementation note:</b> Adapted from
+     * <a href="http://aosong.com/userfiles/files/software/AHT20-21%20DEMO%20V1_3(1).rar">datasheet reference</a>
+     *
+     * @param registry the registry
+     * @see <a href="http://aosong.com/userfiles/files/software/AHT20-21%20DEMO%20V1_3(1).rar">datasheet reference</a>
+     */
+    private void resetRegistry(byte registry) {
+        i2c.writeByte(WRITE_COMMAND);
+        i2c.writeBytes(registry, (byte) 0x00, (byte) 0x00);
+
+        SleepUtil.sleepMillis(5);
+        i2c.writeBytes(READ_COMMAND);
+        final byte[] resp = new byte[3];
+        i2c.readBytes(resp);
+
+        SleepUtil.sleepMillis(10);
+        i2c.writeByte(WRITE_COMMAND);
+        final byte firstByte = (byte) (0xB0 | toUnsignedInt(registry));
+        i2c.writeBytes(firstByte, resp[1], resp[2]);
+    }
+
+    private byte[] readMeasurement(boolean withCrc) {
+        i2c.writeBytes(MEASURE_COMMAND);
+
+        /*
+        Wait 80ms for the measurement to be completed, if the read status word Bit[7] is 0, it means the measurement
+        is completed, and then six bytes can be read continuously; otherwise, continue to wait.
+         */
+        SleepUtil.sleepMillis(80);
+        while ((i2c.readByte() & 0x80) == 0x80) { //first bit of byte = 1 => device busy
+            Logger.debug("Device busy");
+            SleepUtil.sleepMillis(10);
+        }
+
+        /*
+        After receiving six bytes, the next byte is CRC check data, which the user can read as needed. If the receiver
+        needs CRC check, it will send an ACK reply after receiving the sixth byte, otherwise it will send a NACK reply.
+         */
+
+        i2c.writeByte(READ_COMMAND);
+        // 1st status byte,
+        // then 5 bytes with hum & temp,
+        // last byte crc (optional)
+        final byte[] resp = new byte[withCrc ? 7 : 6];
+        i2c.readBytes(resp);
+
+        return resp;
+    }
+
+    private float[] parseResponse(byte[] resp) {
+        if (resp.length > 6) { // check last byte CRC
+            final byte[] data = new byte[6];
+            System.arraycopy(resp, 0, data, 0, data.length);
+
+            final int computedCrc = computeCrc(data);
+            final int expectedCrc = toUnsignedInt(resp[6]);
+
+            if (computedCrc != expectedCrc) {
+                Logger.debug("CRC check failed");
+                return new float[]{-1F, -1F};
+            }
+        }
+
+        // skip status
+        // humidity is next with 20 bits
+        final int hum = (toUnsignedInt(resp[1]) << 16 | toUnsignedInt(resp[2]) << 8 | toUnsignedInt(resp[3])) >> 4;
+        // temperature is next with 20 bits
+        final int temp = (toUnsignedInt(resp[3]) & 0x0F) << 16 | toUnsignedInt(resp[4]) << 8 | toUnsignedInt(resp[5]);
+
+        // magic formula from datasheet
+        final float relativeHum = hum * 100F / 1024 / 1024;
+        final float temperature = temp * 200F / 1024 / 1024 - 50;
```

Comment:
Might want to put some parenthesis around relevant parts as it's hard to parse (and relying on "default" order of operations can bite you).

Reply:
The formulas in the datasheet are:
`RH = raw / (2^20) * 100`
`T = raw / (2^20) * 200 - 50`

For me, it looks readable as it is, but leave me a reply with how you want them / what you think is better or more readable and I will update the PR. It's not clear to me where you want the parantheses.

Reply:
I think the _visual_ issue is you've broken up the `Math.pow(1024,2)` into two divide operations. Yes, the formulas as presented in the datasheet are clearer, but you didn't _quite_ follow it. I would also have made the numeric parts constants, with appropriate references to the formula, just because that's generally good practice.